eng/psakefile.ps1

```diff
@@ -25,7 +25,21 @@ properties {
 Include settings.ps1
 Include utils.ps1
 
-Task default -Depends parser, generator, jit, run-examples
+Task default -Depends checks, parser, generator, jit, run-examples
+
+Task checks -Depends cargo-fmt, cargo-clippy
+
+Task cargo-fmt {
+    Invoke-LoggedCommand -workingDirectory $repo.root -errorMessage "Please run 'cargo fmt --all' before pushing" {
+        cargo fmt --all -- --check
+    }
+}
+
+Task cargo-clippy -Depends init {
+    Invoke-LoggedCommand -workingDirectory $repo.root -errorMessage "Please fix the above clippy errors" {
+        cargo clippy --workspace --all-targets --all-features -- -D warnings 
```

Comment:
It seems like any Clippy lints are treated as errors when building locally. I am using Rust 1.57 and Clippy apparently has an additional lint, and the build fails for me when I run `build.ps1`. IMO, it's better to keep warnings as warnings locally, and only elevate them to errors in CI. That means that minor issues during development can be deferred to later without requiring immediate attention, but the code merged to main is still warning-clean.

Reply:
Specifically, in the packages' lib.rs files: https://github.com/qir-alliance/pyqir/blob/19df229766d6b3dc0efc986e7a030a30bc88461a/qirlib/src/lib.rs#L4

I think this `deny` should be changed to `warn`, and the scripts should set it back to deny with a command-line option only during CI.